pkg/handler/user_email_changed.go

```
@@ -23,23 +24,36 @@ func NewUserEmailChanged(s pkg.UserStore) *UserEmailChanged {
 
 // Handle is the user email changed message handler.
 func (u *UserEmailChanged) Handle(ctx context.Context, m *message.Message) error {
-	defer m.Ack(true)
-
 	user := new(pkg.User)
 	if err := json.Unmarshal(m.Body, user); err != nil {
 		log.Printf("failed to unmarshal message body: %v", err)
+		if err := m.Ack(false); err != nil {
+			log.Printf("failed to ack message: %v", err)
+			return err
+		}
 		return err
 	}
 
-	if err := u.store.Save(user); err != nil {
-		if err == pkg.ErrNotFound {
-			log.Print("user not found")
-		} else {
-			log.Printf("failed to save user to store: %v", err)
+	err := u.store.Save(user)
+
+	switch err {
+	case nil:
+		log.Print("user status successfully changed")
+		if err := m.Ack(false); err != nil {
+			return fmt.Errorf("failed to ack message: %v", err)
+		}
+		return nil
+	case pkg.ErrNotFound:
+		log.Print("user not found")
+		if err := m.Ack(false); err != nil {
+			return fmt.Errorf("failed to ack message: %v", err)
```

let's return the actual error e.g `err` and not `Ack`